.generator/src/generator/templates/model_simple.j2

```diff
@@ -249,33 +249,14 @@ func (o *{{ name }}) Unset{{ propertyName }}() {
 
 func (o {{ name }}) MarshalJSON() ([]byte, error) {
 	toSerialize := {% if model.type == "array" %}make([]interface{}, len(o.Items)){% else %}map[string]interface{}{}{% endif %}
-{#-
-	{{#parent}}
-	{{^isMap}}
-	{{^isArray}}
-	serialized{{parent}}, err{{parent}} := json.Marshal(o.{{parent}})
-	if err{{parent}} != nil {
-		return []byte{}, err{{parent}}
-	}
-	err{{parent}} = json.Unmarshal([]byte(serialized{{parent}}), &toSerialize)
-	if err{{parent}} != nil {
-		return []byte{}, err{{parent}}
-	}
-	{{/isArray}}
-	{{/isMap}}
-	{{#isArray}}
 	if o.UnparsedObject != nil {
 		return json.Marshal(o.UnparsedObject)
 	}
+{%- if model.type == "array" %}
 	for i, item := range o.Items {
 		toSerialize[i] = item
```

Comment:
Could `Items` be an array of pointers? (we might need to copy `item` before assigning it `toSerialize`)

Reply:
What would we need to? We're not modifying them, do we?

Reply:
Yes but if `item` is a pointer, then after the loop ends, all `toSerialize` items will point to the last item.

Reply:
I was not able to get a pointer in there, so I guess we should be fine.